docs/design/features/byreflike-generics.md

```diff
@@ -112,21 +114,21 @@ Adding `gpAcceptByRefLike` to the metadata of a Generic parameter will be consid
 
 Enumerating of constructors/methods on `Span<T>` and `ReadOnlySpan<T>` may throw `TypeLoadException` if `T` is a ByRefLike type. See "Troublesome APIs" above for the list of APIs that cause this condition.
 
-## Special IL Sequences
+## <a name="special_il_sequences"></a> Special IL Sequences
 
-The following are IL sequences involving the `box` instruction. They are used for common C# language constructs and shall continue to be valid, even with ByRefLike types, in cases where the result can be computed at JIT time and elided safely. These sequences must now be elided when the target type is ByRefLike. The conditions where each sequence is elided are described below and each condition will be added to the ECMA-335 addendum.
+The following are IL sequences involving the `box` instruction. They are used for common C# language constructs and shall continue to be valid, even with ByRefLike types, in cases where the result can be computed at JIT compile time or interpretation and elided safely. These sequences **must** now be elided when the target type is ByRefLike. The conditions where each sequence is elided are described below and each condition will be added to the ECMA-335 addendum.
 
-`box` ; `unbox.any` &ndash; The box target type is equal to the unboxed target type.
+`box` ; `unbox.any` &ndash; Becomes a NOP, if the box target type is equal to the unboxed target type.
 
-`box` ; `br_true/false` &ndash; The box target type is non-`Nullable<T>`.
+`box` ; `br_true/false` &ndash; Becomes the constant `true`, if the box target type is non-`Nullable<T>`.
 
-`box` ; `isinst` ; `unbox.any` &ndash; The box, `isint`, and unbox target types are all equal.
+`box` ; `isinst` ; `unbox.any` &ndash; Becomes a NOP, if the box, `isinst`, and unbox target types are all equal.
```

Comment:
I don't understand how these sequences affect the stack state. For example, if Roslyn does

```C#
void M<T>(T t) {
  if (t is Span<int> s) {
}
}
```

That's gonna be a `box; isinst; unbox.any`, but the key thing is that the stack started with a `T` on top, and now it should have `Span<int>`. Otherwise you couldn't legally do a `stloc` to a `Span<int>` local.

Reply:
It is a NOP, the pushed state remains. Yes, it will remain `T`.

Reply:
In that case it's not clear to me why we're doing this. What scenario does this enable?

Reply:
This was an existing recognized sequence that is optimized, no new. It isn't unreasonable if we'd like to skip documenting it at this time.